redbot/core/bot.py

```diff
@@ -1135,10 +1135,14 @@ async def _pre_connect(self) -> None:
                 python_version_changed = True
         else:
             if self._cli_flags.no_cogs is False:
-                packages.extend(await self._config.packages())
+                packages.update(dict.fromkeys(await self._config.packages()))
 
             if self._cli_flags.load_cogs:
-                packages.extend(self._cli_flags.load_cogs)
+                packages.update(dict.fromkeys(self._cli_flags.load_cogs))
+            if self._cli_flags.unload_cogs:
+                for x in self._cli_flags.unload_cogs:
+                    if x in self._cli_flags.unload_cogs:
+                        del packages[x]
```

Comment:
- you've done an `if x in self._cli_flags.unload_cogs` check while you probably wanted to check if a package is in `packages` instead
    - I suggest using `packages.pop(package, None)` instead of the whole if
- `x` isn't a good variable name, use `package` instead
```suggestion
                for package in self._cli_flags.unload_cogs:
                    packages.pop(package, None)
```